spec/v0.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Component used by privileged code to store `DID : CID` assocations. It is
+ * `n:m` relationship (Specific did will have many CAR CIDs associated with it
+ * likewise same CAR CID may be associated with more than one DID)
+ */
+export interface StorageProvider {
+  /**
+   * Upload service will call this once it verified the UCAN in order to associate a
+   * CAR link with a given DID. Upload service is unaware if given `DID` is associated
+   * with some account so it is `StoreProvider`s responsibility to return
+   * `UnknownDIDError` if that is at the case.
+   *
+   * @param group - DID of the group to which car will be added, should be
+   * linked with some account.
+   * @param link - CID of the CAR that user wants to add.
+   * @param proof - CID of the invocation UCAN.
+   */
+  add(
+    group: DID,
+    link: Link,
+    proof: Link
+  ): Result<AddStatus, UnknownDIDError | QuotaViolationError>
+  /**
+   * @param group - DID of the group from wich link needs to be deassociated.
+   * @param link - CID of the CAR that user wants to remove.
+   * @param proof - CID of the invocation UCAN.
+   */
+  remove(
+    group: DID,
+    link: Link,
+    proof: Link
+  ): Result<null, UnknownDIDError | DoesNotHasError>
+}
+ 
+/**
+ * Component used to register accounts and/or link/unlink new DIDs with them.
+ */
+export interface AccessProvider {
```

Comment:
@hugomrdias this was originally defined to be called from trusted code that already performed necessary validations. However now I'm thinking it may make more sense to define this in terms of UCAN RPC and expose as such.

Reply:
Ah we probably need to add `validate` to this interface